component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/voiceotp/VoiceOTPAuthenticator.java

```diff
@@ -0,0 +1,2037 @@
+        content = content.replace(decodedMobileNo, screenValue);
+        content = maskConfiguredValues(context, content);
+        context.setProperty(VoiceOTPConstants.ERROR_INFO, content);
+
+        String errorContent = content;
+        if (log.isDebugEnabled()) {
+            errorContent = contentRaw;
+        }
+        log.error(String.format("Following Error occurred while sending call for user: %s, %s", context
+                .getProperty(VoiceOTPConstants.USER_NAME), errorContent));
+
+        return content;
+    }
+
+    /**
+     * This will mask the configured values.
+     *
+     * @param context
+     * @param content
+     * @return return masked value
+     */
+    private String maskConfiguredValues(AuthenticationContext context, String content) {
+
+        String valuesToMask = context.getAuthenticatorProperties().get(VoiceOTPConstants
+                .VALUES_TO_BE_MASKED_IN_ERROR_INFO);
+        if (StringUtils.isNotEmpty(valuesToMask)) {
+            String[] values = valuesToMask.split(MASKING_VALUE_SEPARATOR);
+            for (String val : values) {
+                content = content.replaceAll(val, getMaskedValue(context, val, 0));
+            }
+
+        }
+        return content;
+    }
+
+    /**
+     * Convert error input stream to a string.
+     *
+     * @param errorStream
+     * @return
+     * @throws IOException
+     */
+    private String readContent(InputStream errorStream) throws IOException {
+
+        BufferedReader br = new BufferedReader(new InputStreamReader(errorStream));
+        StringBuilder sb = new StringBuilder();
+        String output;
+        while ((output = br.readLine()) != null) {
+            sb.append(output);
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Proceed with Voice API's rest call.
+     *
+     * @param context      the AuthenticationContext
+     * @param voiceUrl       the voiceUrl
+     * @param httpMethod   the httpMethod
+     * @param headerString the headerString
+     * @param payload      the payload
+     * @param httpResponse the httpResponse
+     * @param mobile       the mobile number
+     * @param otpToken     the OTP token
+     * @return true or false
+     * @throws IOException
+     * @throws AuthenticationFailedException
+     */
+    public boolean sendRESTCall(AuthenticationContext context, String voiceUrl, String httpMethod,
+                                String headerString, String payload, String httpResponse, String mobile,
+                                String otpToken) throws IOException, AuthenticationFailedException {
+
+
+        log.debug("Preparing message for sending out");
+
+        HttpURLConnection httpConnection;
+        boolean connection;
+        String receivedMobileNumber = URLEncoder.encode(mobile, CHAR_SET_UTF_8);
+
+        if (isOTPNumberSplitEnabled(context)) {
+            otpToken = splitAndEncodeNumber(otpToken, getDivisor(context), context);
```

Comment:
Shall we use a new variable like `formattedOTP` here and preserve the original OTP token?